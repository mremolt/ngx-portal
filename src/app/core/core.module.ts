import { CommonModule } from '@angular/common';
import { InjectionToken, NgModule } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';
import { StoreRouterConnectingModule } from '@ngrx/router-store';
import { ActionReducerMap, META_REDUCERS, MetaReducer, StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

import { environment } from '../../environments/environment';
import { AppEffects } from '../app.effects';
import { RootState, metaReducers, reducers } from '../reducers';

export const REDUCER_TOKEN = new InjectionToken<ActionReducerMap<RootState>>('Registered Reducers');

export function getReducers() {
  return reducers;
}

export function getMetaReducers(): MetaReducer<RootState>[] {
  return metaReducers;
}

@NgModule({
  imports: [
    CommonModule,
    // ngrx
    StoreModule.forRoot(REDUCER_TOKEN),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    EffectsModule.forRoot([AppEffects]),
    StoreRouterConnectingModule.forRoot(),
  ],
  declarations: [],
  providers: [
    {
      provide: REDUCER_TOKEN,
      useFactory: getReducers,
    },
    {
      provide: META_REDUCERS,
      useFactory: getMetaReducers,
    },
  ],
})
export class CoreModule {}
