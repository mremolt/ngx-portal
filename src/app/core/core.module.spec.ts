import { CoreModule, getMetaReducers, getReducers } from './core.module';

describe('CoreModule', () => {
  let coreModule: CoreModule;

  beforeEach(() => {
    coreModule = new CoreModule();
  });

  it('should create an instance', () => {
    expect(coreModule).toBeTruthy();
  });

  describe('getReducers', () => {
    it('returns the reducers', () => {
      expect(getReducers()).toMatchSnapshot();
    });
  });

  describe('getMetaReducers', () => {
    it('returns an Array of meta reducers', () => {
      const subject = getMetaReducers();
      expect(subject).toEqual(expect.any(Array));

      expect(subject).toMatchSnapshot();
    });
  });
});
