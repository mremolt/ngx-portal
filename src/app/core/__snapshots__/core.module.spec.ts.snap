// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreModule getMetaReducers returns an Array of meta reducers 1`] = `
Array [
  [Function],
]
`;

exports[`CoreModule getReducers returns the reducers 1`] = `
Object {
  "router": [Function],
}
`;
